### PR TITLE
feat: adds event overview as separate components

### LINKED ISSUE #76: Add event overview to events route (UI only)
- [x] feature request

**implement Overview(UI) to event/view/index.hbs**

screenshot from 2017-05-24 13-25-03
- Located at `events/<event_id>/`
- Event Sponsors, Manage Roles, General Info and Event setup checklist should be implemented as separate components.

### PR DESCRIPTION
**Pull request by anu-007**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
adds the overview of an event like Event Sponsors, Manage Roles, General Info and Event setup as separate ember component.

#### Changes proposed in this pull request:

alt text

Fixes #76

### CODE REVIEW

**Comment by anu-007** [comment]
@niranjan94 please review.

File: app/templates/components/events/view/overview/event-setup-checklist.hbs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      <div class="right floated content">
+        <div class="ui button"><i class="edit icon"></i>Edit</div>
+      </div>
+      <b class="header">{{t 'Basic Details'}}</b>
+      <div class="description">{{t 'Basic information such as title,date,venue,etc.'}}</div>
+    </div>
+  </div>
+  <div class="ui grey inverted segment">
+    <i class="large checkmark middle aligned icon"></i>
+    <div class="content">
```
Comment (geekyd) [inline app/templates/components/events/view/overview/event-setup-checklist.hbs]: Use grid UI for checklist item,use rows & columns. https://semantic-ui.com/collections/grid.html

**Comment by anu-007** [comment]
@niranjan94 I have done some changes as you suggested below is the new screenshot-
alt text

@geekyd I have used grid layout as a wrapper for the components please check app/templates/events/view/index.hbs and app/templates/events/view.hbs files .thanks !!

File: app/templates/components/events/view/overview/event-setup-checklist.hbs [hunk 2]
```diff
@@ -0,0 +1,56 @@
+<div class="content">
+  <div class="header">{{t 'Event setup checklist'}}</div>
+</div>
+<div class="ui segments content">
+  <div class="ui teal inverted segment">
+    <i class="large checkmark middle aligned icon left floated"></i>
+    <div class="content">
+      <div class="right floated content">
+        <div class="ui button"><i class="edit icon"></i>Edit</div>
```
Comment (niranjan94) [inline app/templates/components/events/view/overview/event-setup-checklist.hbs]: Use a `ui basic button`. Looks better with a colored background.

File: app/templates/components/events/view/overview/event-setup-checklist.hbs [hunk 3]
```diff
@@ -0,0 +1,56 @@
+<div class="content">
+  <div class="header">{{t 'Event setup checklist'}}</div>
+</div>
+<div class="ui segments content">
+  <div class="ui teal inverted segment">
+    <i class="large checkmark middle aligned icon left floated"></i>
```
Comment (niranjan94) [inline app/templates/components/events/view/overview/event-setup-checklist.hbs]: This checkmark should be bigger to take up the full height of the segment

Comment (anu-007) [inline app/templates/components/events/view/overview/event-setup-checklist.hbs]: should I use big icon or huge icon, I tried using huge icon for check mark but it appears very big related to the content.

Comment (niranjan94) [inline app/templates/components/events/view/overview/event-setup-checklist.hbs]: You can use css to adjust this.

**Comment by anu-007** [comment]
alt text

@niranjan94 please review changes.thanks !!

**Comment by niranjan94** [comment]
@anu-007 the event checklist still seems off. Can you use https://semantic-ui.com/elements/step.html instead for it ? (no need to color the background. The default semantic styling should do. And no need edit button either. The steps are clickable)

**Comment by anu-007** [comment]
alt text

I used the following code for each steps-

```javascript
<div class="ui vertical steps">
  <div class="link completed step">
    <div class="content">
      <div class="title">{{t 'Basic Details'}}</div>
      <div class="description">{{t 'Basic information such as title,date,venue,etc.'}}</div>
    </div>
  </div>
```
@niranjan94 please review.thanks!!

File: app/templates/components/events/view/overview/event-setup-checklist.hbs [hunk 4]
```diff
@@ -0,0 +1,36 @@
+<div class="content">
+  <div class="header">{{t 'Event setup checklist'}}</div>
+</div>
+<div class="ui vertical steps">
+  <div class="link completed step">
```
Comment (niranjan94) [inline app/templates/components/events/view/overview/event-setup-checklist.hbs]: Add appropriate icons for all this. See the one's at `/create`

File: app/templates/components/events/view/overview/event-setup-checklist.hbs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    <div class="content">
+      <div class="title">{{t 'Basic Details'}}</div>
+      <div class="description">{{t 'Basic information such as title,date,venue,etc.'}}</div>
```
Comment (niranjan94) [inline app/templates/components/events/view/overview/event-setup-checklist.hbs]: Description => `Basic information including title, date, venue, ticketing`

File: app/templates/components/events/view/overview/event-setup-checklist.hbs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  <div class="link completed step">
+    <div class="content">
+      <div class="title">{{t 'Sponsors'}}</div>
+      <div class="description">{{t 'Details and logos of sponsors'}}</div>
+    </div>
+  </div>
+  <div class="link completed step">
+    <div class="content">
+      <div class="title">{{t 'Session,Tracks,Rooms'}}</div>
+      <div class="description">{{t 'Track details,colors,room locations and session types'}}</div>
```
Comment (niranjan94) [inline app/templates/components/events/view/overview/event-setup-checklist.hbs]: Description => `Tracks details and colors, session lengths, room locations`

File: app/templates/components/events/view/overview/event-setup-checklist.hbs [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  </div>
+  <div class="link completed step">
+    <div class="content">
+      <div class="title">{{t 'Sponsors'}}</div>
+      <div class="description">{{t 'Details and logos of sponsors'}}</div>
+    </div>
+  </div>
+  <div class="link completed step">
+    <div class="content">
+      <div class="title">{{t 'Session,Tracks,Rooms'}}</div>
```
Comment (niranjan94) [inline app/templates/components/events/view/overview/event-setup-checklist.hbs]: Title => `Session, Tracks & Microlocations`

File: app/templates/components/events/view/overview/general-info.hbs [hunk 8]
```diff
@@ -0,0 +1,49 @@
+<div class="content">
+  <div class="header">{{t 'General info'}}</div>
+</div>
+<div class="content">
+  <div class="header"> {{t 'Your event is not live yet. Make it live now'}}</div>
+</div>
+<div class="content">
+  <table class="ui striped table">
```
Comment (niranjan94) [inline app/templates/components/events/view/overview/general-info.hbs]: Use `ui very basic table`

File: app/templates/components/events/view/overview/general-info.hbs [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    <tbody>
+      <tr>
+        <th>{{t 'No. of Revisions'}}</th>
```
Comment (niranjan94) [inline app/templates/components/events/view/overview/general-info.hbs]: Using a `th` inside a `tr` along with a `td`, results in bad styling. Instead make this and use `strong` to make the text bold

File: app/templates/components/events/view/overview/general-info.hbs [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+      <tr>
+        <th>{{t 'End time'}}</th>
+        <td>5:00 PM May 11, 2014</td>
+      </tr>
+      <tr>
+        <th>{{t 'Session types'}}</th>
+        <td>Tech</td>
+      </tr>
+      <tr>
+        <th>{{t 'links and Social Media'}}</th>
```
Comment (niranjan94) [inline app/templates/components/events/view/overview/general-info.hbs]: `L` should be uppercase.

File: app/templates/components/events/view/overview/event-setup-checklist.hbs [hunk 11]
```diff
@@ -0,0 +1,36 @@
+<div class="content">
+  <div class="header">{{t 'Event setup checklist'}}</div>
+</div>
+<div class="ui vertical steps">
```
Comment (anu-007) [inline app/templates/components/events/view/overview/event-setup-checklist.hbs]: I have put the header in a content div and all the steps inside ui vertical steps in separate content div.

**Comment by anu-007** [comment]
@niranjan94 please review.thanks!!

**Comment by anu-007** [comment]
@niranjan94 I have done some changes as you suggested please review.

File: app/templates/components/events/view/overview/event-setup-checklist.hbs (hunk 11 above)
Comment (niranjan94) [inline app/templates/components/events/view/overview/event-setup-checklist.hbs]: The steps still don't seem to be in a content div ?

**Comment by anu-007** [comment]
@niranjan94 please review changes.thanks!!

**Comment by niranjan94** [comment]
@anu-007 please check your commits.

**Comment by anu-007** [comment]
@niranjan94 improved commits please review.thanks!!

**Comment by anu-007** [comment]
@niranjan94 please review.

**Pull request merged**